src/Provider/DebugServiceProvider.php

```diff
@@ -92,91 +93,47 @@ public function register(Application $app)
             });
         }
 
-        $app['debug.handlers_listener'] = $app->share(
-            function ($app) {
-                $errorLevels = $app['config']->get(
-                    $app['debug'] ?
-                    'general/debug_error_level' :
-                    'general/production_error_level'
-                );
-
-                return new DebugHandlersListener(
-                    null, // HttpKernel or Console App exception handler
-                    $app['logger'],
-                    $errorLevels, // Levels
-                    $errorLevels, // Throw at
-                    true, // Scream
-                    $app['code.file_link_format']
-                );
+        // Thrown errors in an integer bit field of E_* constants
+        $app['error_handler.throw_at'] = function ($app) {
+            if ($app['debug']) {
+                return $app['config']->get('general/debug_error_level', E_ALL);
+            } else {
+                return $app['config']->get('general/production_error_level', 0);
             }
-        );
+        };
+
+        // Logged errors in an integer bit field of E_* constants.
+        $app['error_handler.log_at'] = E_ALL; // TODO Should this be the same as throw_at? Previously it was via config env_error_level.
```

Comment:
`env_error_level` because it could be `debug_error_level` or `production_error_level`.

I think they are separate too, but I'm fine changing the default to use the same config value.